ext/RMagick/rmimage.c

```diff
@@ -12796,36 +14038,48 @@ Image_store_pixels(VALUE self, VALUE x_arg, VALUE y_arg, VALUE cols_arg
     size = (long)(cols * rows);
     rm_check_ary_len(new_pixels, size);
 
+    exception = AcquireExceptionInfo();
+#if defined(IMAGEMAGICK_7)
+    okay = SetImageStorageClass(image, DirectClass, exception);
+    CHECK_EXCEPTION()
+#else
     okay = SetImageStorageClass(image, DirectClass);
     rm_check_image_exception(image, RetainOnError);
+#endif
     if (!okay)
     {
+        (void) DestroyExceptionInfo(exception);
```

Comment:
This looks like maybe another fix for an existing bug that should be pulled out into another PR.

Reply:
Not a fix for an existing bug I just pushed a patch to make it more readable